Resources/Prototypes/Entities/Constructible/Power/Engines/PA/particles.yml

```diff
@@ -20,7 +20,8 @@
           !type:PhysShapeAabb
             bounds: "-0.48,-0.48,0.48,0.48"
         hard: false
-        layer: [None]
-        mask:
-        - MobMask
+        layer:
+          # Has to hit mobs, singularity, and singularity generator
+          - MobMask
+        mask: [None]
```

Comment:
This isn't set up right. The particles go through walls with this. I tested and to fix it you should set `layer` to `[None]` and `mask` to `- Opaque`.

Reply:
While this does keep the singularity working, I tested with some reinforced walls and regular walls. The particles seem to go right through both. I don't know if this is really a problem though.

Reply:
Nope, nevermind, the singularity dissolved.
To be clear, you meant, in `particles.yml`, the following:
```
        hard: false
        layer: [None]
        mask:
          - Opaque
```
correct? as this is what I tested with.

Reply:
To clarify what happened with that test:
Singularity successfully formed, but particles ignored both walls and the singularity once formed.
The result was that the singularity dissolved because it couldn't be fed particles.

Reply:
You're going to want to test with a low wall with a window on top, which it should go through, and a solid wall, which it should not go through.

Reply:
Maybe add Opaque to the Singularity layer if it doesn't have that.

Reply:
Alright, just learned that the particles are supposed to go through everything, so MobMask should be fine. You really only need Impassable and Opaque collision since that's what the Singularity and Singularity Generators have for their layers though.

Reply:
At this point I'm confused as to what's what, so please just give me a diff